redisson/src/main/java/org/redisson/connection/SentinelConnectionManager.java

```diff
@@ -373,23 +375,32 @@ public void accept(Object t, Throwable u) {
         };
         
         RFuture<InetSocketAddress> masterFuture = connection.async(StringCodec.INSTANCE, RedisCommands.SENTINEL_GET_MASTER_ADDR_BY_NAME, cfg.getMasterName());
-        masterFuture.onComplete((master, e) -> {
-            if (e != null) {
-                return;
-            }
-
-            RedisURI current = currentMaster.get();
-            RedisURI newMaster = toURI(master.getHostString(), String.valueOf(master.getPort()));
-            if (!newMaster.equals(current)
-                    && currentMaster.compareAndSet(current, newMaster)) {
-                RFuture<RedisClient> changeFuture = changeMaster(singleSlotRange.getStartSlot(), newMaster);
-                changeFuture.onComplete((res, ex) -> {
-                    if (ex != null) {
-                        currentMaster.compareAndSet(newMaster, current);
+        BiConsumer<InetSocketAddress, Throwable> masterListener = new BiConsumer<InetSocketAddress, Throwable>() {
+            @Override
+            public void accept(InetSocketAddress master, Throwable e) {
+                if (e != null) {
+                    if (e instanceof RedisTimeoutException) {
+                        hasConnectionException.set(true);
                     }
-                });
+                    return;
+                }
+
+                RedisURI current = currentMaster.get();
+                RedisURI newMaster = toURI(master.getHostString(), String.valueOf(master.getPort()));
+                if ((!newMaster.equals(current) || hasConnectionException.get())
```

Comment:
we can execute this code block only if `!newMaster.equals(current)`

Reply:
Yes you are right. I will close this PR